magenta/music/musicxml_parser.py

```diff
@@ -552,43 +586,59 @@ def _fix_time_signature(self):
     """
     # Compute the fractional time signature (duration / divisions)
     # Multiply divisions by 4 because division is always parts per quarter note
-    fractional_time_signature = Fraction(self.duration,
-                                         self.state.divisions * 4)
+    numerator = self.duration
+    denominator = self.state.divisions * 4
+    fractional_time_signature = Fraction(numerator, denominator)
 
     if self.state.time_signature is None and self.time_signature is None:
-      # No global time signature yet and no time signature defined
+      # No global time signature yet and no measure time signature defined
       # in this measure (no time signature or senza misura).
       # Insert the fractional time signature as the time signature
       # for this measure
-
       self.time_signature = TimeSignature(self.state)
       self.time_signature.numerator = fractional_time_signature.numerator
       self.time_signature.denominator = fractional_time_signature.denominator
       self.state.time_signature = self.time_signature
     else:
-      # If no global time signature yet, set it to the
-      # time signature in this measure
-      if self.state.time_signature is None:
-        self.state.time_signature = self.time_signature
+
+      fractional_state_time_signature = Fraction(
+          self.state.time_signature.numerator,
+          self.state.time_signature.denominator)
+
+      # Check for pickup measure. Reset time signature to smaller numerator
+      pickup_measure = False
+      if numerator < self.state.time_signature.numerator:
+        pickup_measure = True
 
       # Get the current time signature denominator
       global_time_signature_denominator = self.state.time_signature.denominator
 
       # If the fractional time signature = 1 (e.g. 4/4),
       # make the numerator the same as the global denominator
-      if fractional_time_signature == 1:
+      if fractional_time_signature == 1 and not pickup_measure:
         new_time_signature = TimeSignature(self.state)
         new_time_signature.numerator = global_time_signature_denominator
         new_time_signature.denominator = global_time_signature_denominator
       else:
         # Otherwise, set the time signature to the fractional time signature
+        # Issue #674 - Use the original numerator and denominator
+        # instead of the fractional one
         new_time_signature = TimeSignature(self.state)
-        new_time_signature.numerator = fractional_time_signature.numerator
-        new_time_signature.denominator = fractional_time_signature.denominator
+        new_time_signature.numerator = numerator
+        new_time_signature.denominator = denominator
+
+        new_time_sig_fraction = Fraction(new_time_signature.numerator,
+                                         new_time_signature.denominator)
+
+        if new_time_sig_fraction == fractional_time_signature:
```

Comment:
This section is to preserve the original form of the time signature, right? If that's the case, should it use `numerator` and `denominator` rather than `fractional_time_signature.numerator` and `fractional_time_signature.denominator`?